app/locations/index.js

```diff
@@ -9,17 +9,18 @@ const {
   setUserLocations,
   checkLocationsLength,
   setLocation,
+  setRegion,
   setAllLocations,
   setRegions,
 } = require('./middleware')
 
 router.use(setUserLocations)
 
 // Define routes
-// todo apply only to PMU journeys
 router.get('/', checkLocationsLength, setRegions, locations)
 router.get('/all', setAllLocations, redirect)
 router.get('/:locationId', setLocation, redirect)
+router.get('/region/prisons/:regionId', setRegion, redirect)
```

Comment:
Although we established the current regions from the API only return prisons I wonder if we do want to add that to the URL or not as the frontend should probably be more flexible and support other _regions_ in the future. So maybe this is just `/region/:regionId`?

Reply:
I used to have `/region/:regionId` and changed it to `/region/prisons` because the endpoint deals only with prisons.

I think the route should describe what it does, but I'm happy to revert the change.

Reply:
Yeah, I do agree. It's just now awkward because the API endpoints don't follow the same pattern.

The other thing that doesn't feel very nice is that we will have `404`s at both levels now and I tend to find that a bad design when building nested URLs within an application. Maybe it would be okay if we redirect those somewhere so they don't just have a dead-end. What do you think?

Basically my view is that you should normally be able to remove portions of URLs and still find meaningful things at those URLs. `/region` and `/region/prisons` now wouldn't contain anything.

Reply:
I reverted to `/regions/:regionId`.

> Basically my view is that you should normally be able to remove portions of URLs and still find meaningful things at those URLs. `/region` and `/region/prisons` now wouldn't contain anything.

This is a bit at odds with the current practice of (ab)using middlewares. We have routes that exist only to set values (and redirect), but they have no actual meaning to users.

I checked and dropping the `prison` suffix solves the issue. When the `regionId` is not set, we still route the user to a valid page.

Reply:
>This is a bit at odds with the current practice of (ab)using middlewares. We have routes that exist only to set values (and redirect), but they have no actual meaning to users.

Yep, you're right in that there's nothing that actually lives _there_. What approach do you normally take for such actions? Would you just handle it within a middleware instead of a controller for that route? Ultimately the route needs some form of middleware/controller defined on it to handle that action.

>I checked and dropping the prison suffix solves the issue. When the regionId is not set, we still route the user to a valid page.

👍

Reply:
I think `/region/:regionId` should display the following page:

![screenshotprison](https://user-images.githubusercontent.com/1130499/86556706-ab60eb80-bf86-11ea-9679-2555aa237a90.png)

As it should `/locations/:locationId`.
If I understand the domain model correctly, moves belong to locations, so the current URL could change to:

```
PREVIOUS: http://localhost:3000/moves/day/2020-07-06/8fadb516-f10a-45b1-91b7-a256196829f9/outgoing
PROPOSAL http://localhost:3000/location/region/<regionId>/moves/?day=2020-07-06&type=outgoing
```

The day and the direction are not paths, but filters for the current view. They could be repurposed as query string arguments.

Notice how the proposed URL drills down into the resources:

```
locations -> regions -> moves
``` 

The structure simplifies retrieving (and link sharing) the current state of the UI. I can bookmark the URL and get back to it or I can share it with a coworker. I can also remove parts of the URL and navigate resources (i.e. `/locations/regions/:regionId` should display a list of all regions).
On the technical side, we can rebuild the entire state of your session from the URL (there's no magic middleware journey to follow).

I understand that things started with the best intention and grew organically to this. I'm not expecting to change _all_ routes now, neither in the near future.
However, I think it's important to discuss it so that we can mitigate some of the drawbacks such as misusing the user session.